mods/dye/init.lua

```diff
@@ -1,71 +1,44 @@
-}
-
--- This collection of colors is partly a historic thing, partly something else
-
-local dyes = {
-	{"white",      "White dye",      {dye=1, basecolor_white=1,   excolor_white=1,      unicolor_white=1}},
-	{"grey",       "Grey dye",       {dye=1, basecolor_grey=1,    excolor_grey=1,       unicolor_grey=1}},
-	{"dark_grey",  "Dark grey dye",  {dye=1, basecolor_grey=1,    excolor_darkgrey=1,   unicolor_darkgrey=1}},
-	{"black",      "Black dye",      {dye=1, basecolor_black=1,   excolor_black=1,      unicolor_black=1}},
-	{"violet",     "Violet dye",     {dye=1, basecolor_magenta=1, excolor_violet=1,     unicolor_violet=1}},
-	{"blue",       "Blue dye",       {dye=1, basecolor_blue=1,    excolor_blue=1,       unicolor_blue=1}},
-	{"cyan",       "Cyan dye",       {dye=1, basecolor_cyan=1,    excolor_cyan=1,       unicolor_cyan=1}},
-	{"dark_green", "Dark green dye", {dye=1, basecolor_green=1,   excolor_green=1,      unicolor_dark_green=1}},
-	{"green",      "Green dye",      {dye=1, basecolor_green=1,   excolor_green=1,      unicolor_green=1}},
-	{"yellow",     "Yellow dye",     {dye=1, basecolor_yellow=1,  excolor_yellow=1,     unicolor_yellow=1}},
-	{"brown",      "Brown dye",      {dye=1, basecolor_brown=1,   excolor_orange=1,     unicolor_dark_orange=1}},
-	{"orange",     "Orange dye",     {dye=1, basecolor_orange=1,  excolor_orange=1,     unicolor_orange=1}},
-	{"red",        "Red dye",        {dye=1, basecolor_red=1,     excolor_red=1,        unicolor_red=1}},
-	{"magenta",    "Magenta dye",    {dye=1, basecolor_magenta=1, excolor_red_violet=1, unicolor_red_violet=1}},
-	{"pink",       "Pink dye",       {dye=1, basecolor_red=1,     excolor_red=1,        unicolor_light_red=1}},
+	white		= { name = "White", html = "#FFFFFF" },
+	grey		= { name = "Grey", html = "#7F7F7F" },
+	dark_grey	= { name = "Dark Grey", html = "#3F3F3F" },
+	black		= { name = "Black", html = "#000000" },
+	violet		= { name = "Violet", html = "#8000FF" },
+	blue		= { name = "Blue", html = "#0000FF" },
+	cyan		= { name = "Cyan", html = "#00FFFF" },
+	dark_green	= { name = "Dark Green", html = "#007F00" },
+	green		= { name = "Green", html = "#00FF00" },
+	yellow		= { name = "Yellow", html = "#FFFF00" },
+	brown		= { name = "Brown", html = "#854C30" },
+	orange		= { name = "Orange", html = "#FF7F00" },
+	red			= { name = "Red", html = "#FF0000" },
+	magenta		= { name = "Magenta", html = "#FF00FF" },
+	pink		= { name = "Pink", html = "#FF7FFF" },
 }
 
 -- Define items
 
-for _, row in ipairs(dyes) do
-	local name = row[1]
-	local description = row[2]
-	local groups = row[3]
-	local item_name = "dye:" .. name
-	local item_image = "dye_" .. name .. ".png"
-	minetest.register_craftitem(item_name, {
-		inventory_image = item_image,
-		description = description,
-		groups = groups
+for key, value in pairs(dye.dyes) do
+	local itemName = "dye:"..key
+
+	minetest.register_craftitem(itemName, {
+		inventory_image = "dye_"..key..".png",
+		description = value.name,
```

Comment:
Add `" dye"` to the description. The item isn't the color green, it's `"Green dye"`